distributed/shuffle/tests/test_graph.py

```diff
@@ -65,18 +71,25 @@ def test_raise_on_sparse_data():
     df = dd.from_pandas(
         pd.DataFrame({"x": pd.array(range(10), dtype="Sparse[float64]")}), npartitions=5
     )
-    with pytest.raises(
-        TypeError, match="p2p does not support sparse data"
-    ), dask.config.set({"dataframe.shuffle.method": "p2p"}):
-        df.shuffle("x")
+    if dd._dask_expr_enabled():
+        df.shuffle("x").optimize()
+    else:
+        with pytest.raises(
+            TypeError, match="p2p does not support sparse data"
+        ), dask.config.set({"dataframe.shuffle.method": "p2p"}):
+            df.shuffle("x")
 
 
 def test_raise_on_non_string_column_name():
     df = dd.from_pandas(pd.DataFrame({"a": range(10), 1: range(10)}), npartitions=5)
-    with pytest.raises(
-        TypeError, match="p2p requires all column names to be str"
-    ), dask.config.set({"dataframe.shuffle.method": "p2p"}):
-        df.shuffle("a")
+
+    if dd._dask_expr_enabled():
+        df.shuffle("a").optimize()
```

Comment:
Why does this not raise?

Reply:
It's still a TODO to make an issue for this and the others to raise properly but haven't gotten to it yet.

Reply:
https://github.com/dask-contrib/dask-expr/pull/951